software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+   --      put jsonFromArray(sMain[sNowDoc]) into tResult
+   --      -- each project can have lots of associated files, so the project file gets its own folder
+   --      create folder tPath & "/" & tFileName
+   --      put tPath & "/" & tFileName & "/" & tFileName & ".how" into tFullName
+   --      put tResult into URL ("file:" & tFullName)
+   --   put "saved" into sMain[sNowDoc][0]["saveStatus"]
+   --   put tFullName into sMain[sNowDoc][0]["filepath"]
+   --   put item -1 of tFullName into tForShow
+   --   replace ".how" with "" in tForShow
+   --   put tForShow into field "filenameLabel" stack "HowstrBeta"
+   --   end if
+end saveAsButtMouseUp
+
+on iconCloseMouseUp
+   put "on closeStackRequest" into xX["write"];    devLog xX
+   if sMain[sNowDoc][0]["saveStatus"] is "unsaved" then
+      answer "Close without saving?" with "Close" or "Cancel"
+      if it is "cancel" then exit iconCloseMouseUp
+   end if
+   quit
+end iconCloseMouseUp
+
+on closeStackRequest
+   -- for whatever reason this only gets sent by the window decoration (at least in Windows where I tested it)
+   -- so this blocks the window decorations from closing any of the stacks
+   -- forces user to use the "X" icons
+end closeStackRequest
+
+on hideButtMouseUp
+   put "on hideButtMouseUp" into xX["write"];    devLog xX
+   -- hides whatever elements are selected 
+   -- but it won't hide the focused node
+   -- shouldn't be able to get this message unless something selected cuz button wouldn't be visible
+   -- changeRequest pAsk,@pMain,pDoc
+   -- x[hide]=#,#,#
+   put sMouseHistory["linkSelected"] into tBlerg["hide"]
+   put sMouseHIstory["nodeSelected"] into item (the number of items of tBlerg["hide"])+1 of tBlerg["hide"]
+   put "on hideButtMouseUp > changeRequest with " & tBlerg["hide"] into xX["write"];    devLog xX
+   changeRequest tBlerg,sMain,sNowDoc -- grapherrific.livecode
+   refreshDraw
+   refreshDisplay
+end hideButtMouseUp
+
+on tooltipSwitchChange
+   put the hilite of button "tooltipSwitch" stack "howstrBeta" into tX
+   put tX into sTooltipVisible
+end tooltipSwitchChange
+
+on busySwitchChange
+   get the hilite of button "busySwitch" stack "howstrBeta"
+   --   put it into sBusy
+   set the cBusy of stack "HowstrBeta" to it
+end busySwitchChange
+
+on caseButtMouseUp
+   -- load up existing case node title if it exists
+   put sMainShort[sNowDoc][2]["title"] into tCurrent
+   if tCurrent is empty then
+      ask "name for case node"
+   else
+      ask "name for case node" with tCurrent
+   end if
+   -- only request a change if the case node title actually changed
+   if it is not tCurrent then
+      put it into tStuff["data"]
+      put "title" into tStuff["tag"]
+      put sNowDoc into tStuff["doc"]
+      put 2 into tStuff["rec"]
+      changeInfo sMain,tStuff -- grapheriffic
+      refreshDraw
+      refreshDisplay
+   end if
+end caseButtMouseUp
+
+on miscRightButtMouseUp
+   -- slide all of the misc nodes one place to the right
```

Comment:
@matthewmaier Perhaps it would be better to use a scrolling group. I imagine the graph could get pretty large and complicated